### PR TITLE
Add pusher and floop automation

### PR DESCRIPTION
**Pull request by TahsinAhmed13**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Floop.java [hunk 1]
```diff
@@ -35,6 +37,18 @@ public boolean isOpen() {
         return floopSolenoid.get();
     }
 
+    public void push() {
+        pusherSolenoid.set(true);
+    }
+
+    public void pull() {
```
Comment (justinkim) [inline src/main/java/frc/robot/subsystems/Floop.java]: It's not quite a pulling action. You might want to reword this. Maybe `retract` or `makeExtremelyTiny`

File: src/main/java/frc/robot/commands/FloopPushCommand.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Add your docs here.
+ */
+public class FloopPushCommand extends InstantCommand {
+  /**
+   * Add your docs here.
+   */
+  public FloopPushCommand() {
+    super();
+    // Use requires() here to declare subsystem dependencies
+    // eg. requires(chassis);
```
Comment (huiminwu) [inline src/main/java/frc/robot/commands/FloopPushCommand.java]: do you not need to specify what subsystem is required for this command

File: src/main/java/frc/robot/commands/FloopCloseCommand.java [hunk 3]
```diff
@@ -18,6 +18,13 @@ public FloopCloseCommand() {
 
     @Override
     protected void initialize() {
-        Robot.floop.close();
+        if(Robot.IRsensor.get()) { 
+            Robot.floop.open();
+            Robot.floop.push();
```
Comment (huiminwu) [inline src/main/java/frc/robot/commands/FloopCloseCommand.java]: there should be a white after you open, before you push

File: src/main/java/frc/robot/commands/FloopOpenCommand.java [hunk 4]
```diff
@@ -19,5 +19,8 @@ public FloopOpenCommand() {
     @Override
     protected void initialize() {
         Robot.floop.open();
+        if(Robot.IRsensor.get()) {
```
Comment (huiminwu) [inline src/main/java/frc/robot/commands/FloopOpenCommand.java]: separate the manual movement from ir sensor movement so we have a fallback when it breaks

File: src/main/java/frc/robot/OI.java [hunk 5]
```diff
@@ -57,19 +61,17 @@ public OI() {
...
+        operatorGamepad.getBottomButton().whenReleased(new FloopStopScoreCommand());
 
         operatorGamepad.getDPadLeft().whenPressed(new LiftToggleCommand());
-        operatorGamepad.getDPadDown().whenPressed(new LiftMoveToHeightCommand(RobotMap.HP_LEVEL_1_HEIGHT));
-        operatorGamepad.getDPadRight().whenPressed(new LiftMoveToHeightCommand(RobotMap.HP_LEVEL_2_HEIGHT));
-        operatorGamepad.getDPadUp().whenPressed(new LiftMoveToHeightCommand(RobotMap.HP_LEVEL_3_HEIGHT));
 
         operatorGamepad.getRightAnalogButton().whenPressed(new AbomToggleCommand());
-        operatorGamepad.getLeftAnalogButton().whenPressed(new LiftSlowToggleCommand());
+        operatorGamepad.getLeftAnalogButton().whileHeld(new LiftSlowToggleCommand());
```
Comment (PrathamRawat) [inline src/main/java/frc/robot/OI.java]: Why was this changed?

File: src/main/java/frc/robot/commands/FloopCloseCommand.java [hunk 6]
```diff
@@ -20,4 +20,16 @@ public FloopCloseCommand() {
     protected void initialize() {
         Robot.floop.close();
     }
+
+    @Override
+    protected void execute() {
```
Comment (PrathamRawat) [inline src/main/java/frc/robot/commands/FloopCloseCommand.java]: Can you add functionality to make a button turn off automation?

File: src/main/java/frc/robot/commands/FloopPushCommand.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+public class FloopPushCommand extends InstantCommand {
+
+  public FloopPushCommand() {
+    super();
+    requires(Robot.floop);
+  }
+
+  @Override
+  protected void initialize() {
+    if (!Robot.floop.pushed()) {
```
Comment (PrathamRawat) [inline src/main/java/frc/robot/commands/FloopPushCommand.java]: You don't need these if statements, it should be fine to remove them

File: src/main/java/frc/robot/commands/FloopStartScoreCommand.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import frc.robot.Robot;
+
+public class FloopStartScoreCommand extends InstantCommand {
+  public FloopStartScoreCommand() {
+    requires(Robot.floop);
+  }
+
+  @Override
+  protected void initialize() {
+    Robot.floop.close();
```
Comment (PrathamRawat) [inline src/main/java/frc/robot/commands/FloopStartScoreCommand.java]: Maybe put a wait here? Talk to engineering to make sure

File: src/main/java/frc/util/Gamepad.java [hunk 9]
```diff
@@ -561,7 +561,7 @@ public boolean get() {
         }
     }
     
-    public void gamepadRumble(double intensity){
+    public void rumble(double intensity){
```
Comment (PrathamRawat) [inline src/main/java/frc/util/Gamepad.java]: Why was this changed here?

File: src/main/java/frc/robot/commands/FloopStartScoreCommand.java (hunk 8 above)
Comment (rmui10) [reply src/main/java/frc/robot/commands/FloopStartScoreCommand.java]: Mr. Blay says that they should happen at the same time.

File: src/main/java/frc/robot/OI.java [hunk 10]
```diff
@@ -67,11 +68,12 @@ public OI() {
         operatorGamepad.getRightButton().whenReleased(new FloopOpenCommand());
         operatorGamepad.getBottomButton().whileHeld(new FloopStartScoreCommand());
         operatorGamepad.getBottomButton().whenReleased(new FloopStopScoreCommand());
+        operatorGamepad.getLeftButton().whenPressed(new FloopAutomationToggleCommand());
 
         operatorGamepad.getDPadLeft().whenPressed(new LiftToggleCommand());
 
         operatorGamepad.getRightAnalogButton().whenPressed(new AbomToggleCommand());
-        operatorGamepad.getLeftAnalogButton().whileHeld(new LiftSlowToggleCommand());
+        operatorGamepad.getLeftAnalogButton().whenPressed(new LiftSlowToggleCommand());
```
Comment (PrathamRawat) [inline src/main/java/frc/robot/OI.java]: Change this as requested by Hui Min
